src/lxc/lxc_user_nic.c

```diff
@@ -117,24 +255,41 @@ static int get_alloted(char *me, char *intype, char *link)
 		return -1;
 	}
 
+	groups = get_groupnames();
 	while ((getline(&line, &len, fin)) != -1) {
-		ret = sscanf(line, "%99[^ \t] %99[^ \t] %99[^ \t] %d", user, type, br, &n);
+		ret = sscanf(line, "%99[^ \t] %99[^ \t] %99[^ \t] %d", name, type, br, &n);
 
 		if (ret != 4)
 			continue;
-		if (strcmp(user, me) != 0)
+
+		if (strlen(name) == 0)
 			continue;
+
+		if (strcmp(name, me) != 0)
+		{
+			if (name[0] != '@')
+				continue;
+			if (!name_is_in_groupnames(name+1, groups))
+				continue;
+		}
 		if (strcmp(type, intype) != 0)
 			continue;
 		if (strcmp(link, br) != 0)
 			continue;
-		free(line);
-		fclose(fin);
-		return n;
+
+		/* found the user or group with the appropriate settings, therefore finish the search.
+		 * what to do if there are more than one applicable lines? not specified in the docs.
+		 * since getline is implemented with realloc, we don't need to free line until exiting func.
+		 */
+		append_alloted(alloted, name, n);
+		count += n;
+		break;
```

Comment:
I'm pretty sure the "count += n;" line only ever gets executed once; given the += here, I assume that means this loop is intended to execute this line potentially several times. Please re-examine this loop to ensure it works as desired. If this line is only expected to execute once, please change to a count = n; to make clear that only a single execution is expected and desired.